hcloud/networks/client.py

```diff
@@ -229,48 +230,66 @@ def create(
                 Array of subnets allocated
         :param routes: List[:class:`NetworkRoute <hcloud.networks.domain.NetworkRoute>`]
                 Array of routes set in this network
+        :param expose_routes_to_vswitch: Optional[bool]
+                Indicates if the routes from this network should be exposed to the vSwitch subnet
         :param labels: Dict[str, str] (optional)
                 User-defined labels (key-value pairs)
         :return: :class:`BoundNetwork <hcloud.networks.client.BoundNetwork>`
         """
         data = {"name": name, "ip_range": ip_range}
         if subnets is not None:
-            data["subnets"] = [
-                {
+            data_subnets = []
+            for subnet in subnets:
+                data_subnet = {
                     "type": subnet.type,
                     "ip_range": subnet.ip_range,
                     "network_zone": subnet.network_zone,
                 }
-                for subnet in subnets
-            ]
+                if subnet.vswitch_id is not None:
+                    data_subnet["vswitch_id"] = subnet.vswitch_id
```

Comment:
According to the docs, the vswitch_id field is required when the subnet type is a vswitch. Should we leave this to the user, or do we want to do some validation in the library ?

Reply:
There is always the risk that the validation gets "out-of-sync" with the API, which causes frustration with the user. I would only add client side validation if it substantially improves the user experience.